src/hayhooks/server/utils/deploy_utils.py

```diff
@@ -389,15 +372,124 @@ def deploy_pipeline_files(
+        options: Optional dict with additional deployment options. Supported keys:
+            - description: Optional[str]
+            - skip_mcp: Optional[bool]
+    """
+
+    # Optionally save YAML to disk as pipelines/{name}.yml (default True)
+    save_file: bool = True if options is None else bool(options.get("save_file", True))
+    if save_file:
+        save_yaml_pipeline_file(pipeline_name, source_code, settings.pipelines_dir)
+
+    # Add pipeline to the registry and build metadata (request/response models)
+    add_yaml_pipeline_to_registry(
+        pipeline_name=pipeline_name,
+        source_code=source_code,
+        overwrite=overwrite,
+        description=(options or {}).get("description"),
+        skip_mcp=(options or {}).get("skip_mcp"),
+    )
+
+    if app:
+        add_pipeline_yaml_api_route(app, pipeline_name)
+
+    return {"name": pipeline_name}
+
+
+def add_yaml_pipeline_to_registry(
+    pipeline_name: str,
+    source_code: str,
+    overwrite: bool = False,
+    description: Optional[str] = None,
+    skip_mcp: Optional[bool] = False,
+) -> None:
+    """
+    Add a YAML pipeline to the registry.
+
+    Args:
+        pipeline_name: Name of the pipeline to deploy
+        source_code: Source code of the pipeline
+    """
+
+    log.debug(f"Checking if YAML pipeline '{pipeline_name}' already exists: {registry.get(pipeline_name)}")
+    if registry.get(pipeline_name):
+        if overwrite:
+            log.debug(f"Clearing existing YAML pipeline '{pipeline_name}'")
+            registry.remove(pipeline_name)
+        else:
+            msg = f"YAML pipeline '{pipeline_name}' already exists"
+            raise PipelineAlreadyExistsError(msg)
+
+    clog = log.bind(pipeline_name=pipeline_name, type="yaml")
+
+    clog.debug("Creating request/response models from declared YAML inputs/outputs")
+
+    # Build request/response models from declared YAML inputs/outputs using resolved IO types
+    try:
+        resolved_io = get_inputs_outputs_from_yaml(source_code)
+
+        pipeline_inputs = resolved_io["inputs"]
+        pipeline_outputs = resolved_io["outputs"]
+
+        # Prefer resolved IO-based flat models for API schema
+        request_model = get_request_model_from_resolved_io(pipeline_name, pipeline_inputs)
+        response_model = get_response_model_from_resolved_io(pipeline_name, pipeline_outputs)
+    except Exception as e:
+        clog.error(f"Failed creating request/response models for YAML pipeline: {e!s}")
+        raise
+
+    metadata = {
+        "description": description or pipeline_name,
+        "request_model": request_model,
+        "response_model": response_model,
+        "skip_mcp": bool(skip_mcp),
+    }
+
+    clog.debug(f"Adding YAML pipeline to registry with metadata: {metadata}")
+
+    # Store the instantiated pipeline together with its metadata
+    # NOTE: We want to create an AsyncPipeline here so we can avoid using
+    #       run_in_threadpool when running the pipeline.
```

Comment:
I think we should make it more clear in docstrings that only `AsyncPipeline` is supported when deploying with yaml. E.g. Use "Haystack AsyncPipeline.run_async" instead of "Haystack Pipeline.run"

Reply:
I've added a Note in the docstring!